### PR TITLE
#167511346: Create Heroku pipelines

### PR DESCRIPTION
**Pull request by fahdjamy**
#### Description

- setup Heroku deployment.

#### Type of change

- [x] setup Heroku
- [x] create pipelines for the app on Heroku.

#### How to Test

- checkout Heroku review apps [Review apps](https://dashboard.heroku.com/pipelines/a2c773c1-fb84-4b92-bb80-53b1396790b1).

[Delivers #167511346]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
   "author": "muhwezi deo <aggrey256@gmail.com>",
   "license": "MIT",
   "scripts": {
-    "start": "tsc && node dist/index.js"
+    "start": "tsc && node dist/index.js",
+    "heroku-prebuild": "npx rimraf package-lock.json npm-shrinkwrap.json node_modules && npm install yarn && yarn install",
```
Comment (MuhweziDeo) [inline package.json]: what is this script

**Review by MuhweziDeo — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (fahdjamy) [reply package.json]: Man, ain't asked your review yet.
Besides, as the name suggests, prebuild installs yarn then yarn installs the packages. yarn is technically lighter than npm hence the script.

File: Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: node myexpress.js
```
Comment (MuhweziDeo) [inline Procfile]: this should point to our entry point

File: package.json (hunk 1 above)
Comment (MuhweziDeo) [reply package.json]: Thanks

File: Procfile (hunk 2 above)
Comment (MuhweziDeo) [reply Procfile]: the one that has been bundled in dist

**Pull request closed without merging**